src/components/Coaching/CoachingDetail/MonthlyCommitment/useMonthlyCommitmentAverage.ts

```diff
@@ -0,0 +1,97 @@
+import { useMemo } from 'react';
+import { DateTime } from 'luxon';
+import { AccountList, Maybe } from 'src/graphql/types.generated';
+import { useMonthlyCommitmentSingleMonthQuery } from './MonthlyCommitment.generated';
+
+type MpdInfo = Pick<AccountList, 'activeMpdStartAt' | 'activeMpdFinishAt'>;
+
+interface MonthlyCommitmentAverageResult {
+  loading: boolean;
+  average: number | null;
+}
+
+/*
+ * Calculate the monthly commitment average for an account list. It represents the amount of net
+ * monthly commitments that the account list has gained per month. For example, if the account list
+ * went from $5000/month committed to $6000/month in 4 months of active MPD, the monthly commitment
+ * average would be $250/month.
+ */
+export const useMonthlyCommitmentAverage = (
+  accountListId: string,
+  mpdInfo: Maybe<MpdInfo>,
+): MonthlyCommitmentAverageResult => {
+  // Calculate the first month that the user was on active MPD
+  const startMonth = useMemo(() => {
+    if (!mpdInfo?.activeMpdStartAt) {
+      return null;
+    }
+
+    const startAt = DateTime.fromISO(mpdInfo.activeMpdStartAt);
+    if (startAt > DateTime.now()) {
+      // If the start at date is in the future, don't load commitment data for the start at date
+      return null;
+    }
+    return startAt.endOf('month');
+  }, [mpdInfo]);
+
+  // Calculate the last month that the user was on active MPD
+  const finishMonth = useMemo(() => {
+    if (!mpdInfo?.activeMpdFinishAt) {
+      return null;
+    }
+
+    const finishAt = DateTime.fromISO(mpdInfo.activeMpdFinishAt);
+    // If the finish at date is in the future, clamp it to the end of the current month
+    return DateTime.min(finishAt, DateTime.now()).endOf('month');
```

Comment:
It looks like you are always clamping it to the end of the month, which could not be in the future. Maybe I'm understanding the comment incorrectly.

Reply:
The goal is to use the end of this month or the end of the finish at month, whichever comes first. It's OK if the finish at month is in the past. But if it's in the future, we want to use the end of the current month to not mess up the average calculation. Do you think the comment could be improved?

Reply:
your comment makes sense, I think I was misunderstanding.